### DISCUSSION TITLE
Split the `attribute` ranking rule into two sub-ranking rules: `field` and `position`.

### DISCUSSION THREAD

**Discussion by loiclec** [meilisearch/product#653] [Feedback & Feature Proposal]
Following https://github.com/meilisearch/meilisearch/issues/3222 , I think it would be a good idea to split the `attribute` ranking rule into two sub-ranking rules.

1. `field` would prioritise documents containing query words in important fields first
2. `position` would prioritise documents containing query words appearing at the beginning of the fields

Note that `field` interacts with the setting `searchableAttributes` and its functionality overlaps with another proposed ranking rule:  [`words in [attribute set]`](https://github.com/meilisearch/product/discussions/595).

The names `field` and `position` probably aren't the best. Other suggestions were `attribute:no-position` and `attribute:position`. 

Note that technically, within the search engine, these two ranking rules are already implemented but not exposed to the user. In fact, the current rule `attribute` corresponds to the rule `field` immediately followed by `position`.

**Comment by NicolasCharpentier** [comment]
Hi 

I am in favor of this as we have in our index data like that 
```
id: 1
field1: Mr. John Smith
field2: 2 street of the Burger king, New York

id: 2
field1: Mr. Smith the third
field2: 3 street of the Burger king, New York

id: 3
field1: Subscription n°2023-122 $200
field2: Paid by Mr. Smith
```

When querying for "Smith", we'd like 
- the two first documents (id 1 and id 2) with the same score, as they both have "Smith" in their primary field
- the last document (id 3) with a lower score

So totally regardless of the position cos the position there really does not matter.

**Comment by NicolasCharpentier** [comment]
Looks like new 1.6 version with it's proximity customization could solve this, not sure tho as my problem was on "implicit" `proximity` rule applied when turning on the `attributes` rule. I'll check 

![image](https://github.com/meilisearch/product/assets/9944995/e9eee5a8-006b-4219-9791-e04c9bb11f3f)

**Reply by ManyTheFish** [reply]
Hello @NicolasCharpentier,
This will not change the behavior of the `Attributes` ranking rule, so no, this feature doesn't help.
However, the feature changes several things related to the proximity between words in the search engine, so changing this setting could make Meilisearch return the results in a better order from your point of view. 😄

**Comment by lswinblad** [comment]
Are there any news on this? :)

**Reply by ManyTheFish** [reply]
Nothing new @lswinblad,
working on the keyword search is not a priority for Meilisearch for now,
sorry.
